### ISSUE TITLE
Poll for updates

### ISSUE THREAD

**Issue by LucianBuzzo** [balena-io/pensieve#10]
<a href="https://github.com/LucianBuzzo"><img src="https://avatars0.githubusercontent.com/u/15064535?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [LucianBuzzo](https://github.com/LucianBuzzo)**
_Tuesday Aug 08, 2017 at 15:08 GMT_
_Originally opened as https://github.com/resin-io/scratchpad/issues/33_

----

We could poll for changes to the currently viewed file and refresh the file if it has changed.
We should warn the user if they are trying to save a file that has changed before their edit started.

**Comment by lurch** [comment]
I assume if (when?) this was added it would also apply to Views etc.? I.e. if user A does a search and gets 5 results, and then user B adds a new entry which also matches this search, then when user A's pensieve polls for updates, it'd update the search to show 6 results?

**Comment by LucianBuzzo** [comment]
@lurch yep this is correct, Any changes to the underlying data model would be reflected almost instantly in the UI

**Comment by lurch** [comment]
So without the automatic update polling, does the user just need to refresh their browser page to get the 'latest' data?

**Comment by LucianBuzzo** [comment]
@lurch yep, if you reload you're page you will get the latest data **that is available from the GitHub API**. In some instances I've seen the GH API lag behind the actual latest commit by some time 😒 
You will also get the latest data if you make an update to the document, as Pensieve will sync data before running an update, to try and make sure that you don't accidentally overwrite anything. If a diff on the entry that you're editing is seen at the point of the sync running, it puts the edit screen into a 'diff mode' where you can manually resolve any changes.
